File: app/src/main/java/com/example/homeworkproject/model/ApiState.java
package com.example.homeworkproject.model;

public class ApiState<T>  {

    public static Status status;

    public ApiState(Status status) {
        ApiState.status = status;
    }

    public static <T> ApiState<T> success(){
        return new ApiState<>(Status.SUCCESS);
    }

    public static <T> ApiState<T> error(){
        return new ApiState<>(Status.ERROR);
    }

    public static <T> ApiState<T> loading(){
        return new ApiState<>(Status.LOADING);
    }
    public static <T> ApiState<T> blank(){
        return new ApiState<>(Status.BLANK);
    }

    public enum Status {
        SUCCESS, ERROR, LOADING, BLANK

    }


}

File: app/src/main/java/com/example/homeworkproject/model/ApiState.kt
package com.example.homeworkproject.model

class ApiState<T>(status: Status?) {
    enum class Status {
        SUCCESS, ERROR, LOADING, BLANK
    }

    companion object {
        var status: Status? = null
        fun <T> success(): ApiState<T> {
            return ApiState(Status.SUCCESS)
        }

        fun <T> error(): ApiState<T> {
            return ApiState(Status.ERROR)
        }

        fun <T> loading(): ApiState<T> {
            return ApiState(Status.LOADING)
        }

        fun <T> blank(): ApiState<T> {
            return ApiState(Status.BLANK)
        }
    }

    init {
        Companion.status = status
    }
}
